backend/spec/requests/api/v1/reset_password_spec.rb

```diff
@@ -21,19 +21,18 @@
         let(:params) { {email: "user@example.com"} }
         let("X-CSRF-TOKEN") { get_csrf_token }
 
-        run_test!
-
-        # head ok so no generate example
-        it "sends email" do
-          mail = ActionMailer::Base.deliveries.last
-          expect(mail.subject).to eq("Reset password instructions")
-          expect(mail.to.first).to eq("user@example.com")
-          expect(user.reload.reset_password_token).to be_present
+        it "sends email" do |example|
+          expect {
+            submit_request example.metadata
+            assert_response_matches_metadata example.metadata
+          }.to have_enqueued_mail(DeviseMailer, :reset_password_instructions).with(user, String, fallback_language: :en)
         end
 
         context "invalid email" do
           let(:params) { {email: "invalid"} }
 
+          run_test!
```

Comment:
I'm sometimes confused when this should be placed ;]

Reply:
When you look into this method, it is in reality just shortcut, which runs:
```
submit_request example.metadata
assert_response_matches_metadata example.metadata
```
as you have already seen, sometimes I run these commands manually so I can wrap them for example with `have_enqueued_mail`, but If you don't trigger them on your own, you need to use `run_test!` so you get API response.